src/fsharp/nameres.fs

```diff
@@ -2559,24 +2575,31 @@ let ResolveCompletionsInType (ncenv: NameResolver) nenv isApplicableMeth m ad st
         result
 
     let pinfoItems = 
+        let pinfos = 
+            match completionTargets with
+            | ResolveCompletionTargets.SettablePropertiesAndFields -> pinfos |> List.filter (fun p -> p.HasSetter)
+            | _ -> pinfos
+
         pinfos
         |> List.map (fun pinfo -> DecodeFSharpEvent [pinfo] ad g ncenv m)
-        |> List.filter (fun pinfo-> match pinfo with
-                                    | Some(Item.Event(einfo)) -> IsStandardEventInfo ncenv.InfoReader m ad einfo
-                                    | _ -> pinfo.IsSome)
+        |> List.filter (fun pinfo->
+            match pinfo with
+            | Some(Item.Event(einfo)) -> completionTargets.ResolveAll && IsStandardEventInfo ncenv.InfoReader m ad einfo
+            | _ -> pinfo.IsSome)
         |> List.map (fun pinfo->pinfo.Value)
 
-    let addersAndRemovers = 
-        pinfoItems 
-        |> List.map (function Item.Event(FSEvent(_,_,addValRef,removeValRef)) -> [addValRef.LogicalName;removeValRef.LogicalName] | _ -> [])
-        |> List.concat
-    
     // REVIEW: add a name filter here in the common cases?
     let minfos = 
-        AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m typ 
-        |> List.filter minfoFilter
-        |> List.filter (fun minfo -> not(addersAndRemovers|>List.exists (fun ar-> ar = minfo.LogicalName)))
-
+        if completionTargets.ResolveAll then
+            let addersAndRemovers = 
+                pinfoItems 
+                |> List.map (function Item.Event(FSEvent(_,_,addValRef,removeValRef)) -> [addValRef.LogicalName;removeValRef.LogicalName] | _ -> [])
+                |> List.concat
+
+            AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m typ 
+            |> List.filter minfoFilter
+            |> List.filter (fun minfo -> not(addersAndRemovers|>List.exists (fun ar-> ar = minfo.LogicalName)))
```

Comment:
@enricosada I don't think that code is clearer nor easier to understand - in fact it's the opposite.  

Also, we don't use "<|" in the F# compiler codebase much - barely at all - again for readability.